### PR TITLE
stricter trusted hosts

### PR DESCRIPTION
**Pull request by STRd6**
## Description

No longer allow arbitrary subdomains of builder.io in trusted hosts.

### CODE REVIEW

File: packages/core/src/builder.class.test.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Builder } from './builder.class';
+
+describe('Builder', () => {
+  test('trustedHosts', () => {
+    expect(Builder.isTrustedHost('example.com')).toBe(false);
+    expect(Builder.isTrustedHost('localhost')).toBe(true);
+    expect(Builder.isTrustedHost('builder.io')).toBe(true);
+    expect(Builder.isTrustedHost('123-review-build.beta.builder.io')).toBe(true);
```
Comment (samijaber) [inline packages/core/src/builder.class.test.ts:8]: can we add a test for the problematic `builder.io` subdomains? `cdn.builder.io`, etc.

Comment (mrkoreye) [reply packages/core/src/builder.class.test.ts:8]: What about white label instances? Such as `umg-mira.builder.io` or  `marketer.mira.umusic.com` etc. I suppose that is why we have the `registerTrustedHost` method? 

Can we add tests to that to make sure it works as expected? My worry is that if we release this as a patch update it might break some weird edge cases.

Comment (mrkoreye) [reply packages/core/src/builder.class.test.ts:8]: One more we probably want to add is `playground.builder.io`

Comment (mrkoreye) [reply packages/core/src/builder.class.test.ts:8]: And `qa.builder.io`, also will this update break support for `www.builder.io`?

Comment (samijaber) [reply packages/core/src/builder.class.test.ts:8]: Should be a minor release, since it potentially involves changes on the user's end to add their hosts

File: packages/core/src/builder.class.test.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Builder } from './builder.class';
+
+describe('Builder', () => {
+  test('trustedHosts', () => {
+    expect(Builder.isTrustedHost('localhost')).toBe(true);
+    expect(Builder.isTrustedHost('builder.io')).toBe(true);
+    expect(Builder.isTrustedHost('beta.builder.io')).toBe(true);
```
Comment (samijaber) [inline packages/core/src/builder.class.test.ts:7]: Another small edge-case to test: `evildomainbeta.builder.io`

**Review by samijaber — APPROVED** [review]
